src/MongoDB.Driver.Core/Core/Misc/RegexExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Text.RegularExpressions;
+
+namespace MongoDB.Driver.Core.Misc
+{
+    internal static class RegexExtensions
+    {
+        public static string ReplaceGroup(this Regex regex, string source, string groupName, string replacement)
```

Comment:
It's a little odd to introduce a new extension method that is only used in one place.

And if we are going to add this method do we need to write tests for it?

Have you tried using `Regex.Replace(input, pattern, "<hidden>")` instead? I assume a suitable pattern could be written and then this method (extension or otherwise) would not be needed.

Reply:
>Regex.Replace(input, pattern, "<hidden>")

yes, I've tried it. It works, but replaces more than just a group content. It also replaces a group brackets like `://` and `@`

I'm not sure that we need to add the tests for almost `built-in` logic. If you think that this extension is not required. I can move this method into `ConnectionString.cs`